code/modules/admin/sql_notes.dm

```diff
@@ -133,15 +160,18 @@
 			output += " <a href='?_src_=holder;shownoteckeylinkless=[target_ckey]'>\[Refresh Page\]</a></center>"
 		output += ruler
 		while(query_get_notes.NextRow())
-			var/id = query_get_notes.item[1]
+			var/secret = text2num(query_get_notes.item[1])
+			if(linkless && secret)
+				continue
```

Comment:
I think that maybe secret notes should just have a generic grayed out "This note is withheld from players." instead, so that the presence of secret notes are known.

Has upsides and downsides, but I personally think it's good to do. Should definitely discuss it though, because maybe the admins will dislike the downsides enough.

I guess then they won't be secret notes, just hidden notes.